src/lib.rs

```diff
@@ -2,50 +2,196 @@
 #![cfg_attr(not(feature = "std"), no_std)]
 
 #[cfg(feature = "std")]
-use serde_derive::{Deserialize, Serialize};
-
 use rstd::prelude::*;
 
-use codec::{Decode, Encode};
-use srml_support::{
-    decl_event, decl_module, decl_storage, dispatch, ensure, StorageMap, StorageValue,
-};
+use runtime_primitives::traits::Zero;
+use srml_support::traits::Currency;
+use srml_support::{decl_module, decl_storage, ensure, StorageMap, StorageValue};
+
+mod mint;
+mod mock;
+mod tests;
 
-// mod mock;
-// mod tests;
+pub use mint::{AdjustCapacityBy, AdjustOnInterval, Adjustment, Mint, MintingError};
 
 use system;
-use system::{ensure_root, ensure_signed};
 
-pub trait Trait: system::Trait + timestamp::Trait + Sized {
-    type Event: From<Event<Self>> + Into<<Self as system::Trait>::Event>;
+pub type MintId = u64;
+pub const FIRST_TOKEN_MINT_ID: MintId = 1;
+
+pub trait Trait: system::Trait + Sized {
+    /// The currency to mint
+    type Currency: Currency<Self::AccountId>;
 }
 
+pub type BalanceOf<T> =
+    <<T as Trait>::Currency as Currency<<T as system::Trait>::AccountId>>::Balance;
+
 decl_storage! {
-    trait Store for Module<T: Trait> as TokenMint {
+    trait Store for Module<T: Trait> as Minting {
```

Comment:
I seemed like a better fit for the purpose of the module, since it represents the module, not an individual mint. Like the staking module is not called Stake